File: tests/adapters/javascript/goldens/function_bodies/large_only_strip.js
/**
 * JavaScript module for testing function body optimization.
 */

class Calculator {
    constructor(name = "default") // … method body omitted (4 lines)

    add(a, b) // … method body omitted (6 lines)

    multiply(a, b) // … method body omitted (5 lines)

    getHistory() {
        return [...this.history];
    }

    #validateInput(value) {
        if (typeof value !== 'number') {
            throw new Error('Input must be a number');
        }

        if (!isFinite(value)) {
            throw new Error('Input must be finite');
        }

        return true;
    }
}

export function processUserData(users) // … function body omitted (13 lines)

// Arrow functions for testing different function types
const simpleArrow = () => "simple";

const complexArrow = (data) => // … arrow_function_body omitted (8 lines);

const asyncArrow = async (url) => // … arrow_function_body omitted (15 lines);

// Function with multiple overloads (via JSDoc)
/**
 * @overload
 * @param {string} value
 * @returns {string}
 */
/**
 * @overload
 * @param {number} value
 * @returns {number}
 */
function overloadedFunction(value) // … function body omitted (7 lines)

// Generic function (via JSDoc)
/**
 * @template T
 * @param {T[]} items
 * @param {(item: T) => T} processor
 * @returns {T[]}
 */
function processArray(items, processor) // … function body omitted (14 lines)

// Default export function
export default function main() // … function body omitted (13 lines)

File: tests/adapters/cpp/goldens/function_bodies/large_only_strip.cpp
/**
 * C++ module for testing function body optimization.
 */

#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

struct User {
    int id;
    std::string name;
    std::string email;
};

struct ProcessingResult {
    std::vector<User> valid;
    std::vector<User> invalid;
};

class Calculator {
private:
    std::vector<std::string> history;
    std::string name;

public:
    Calculator() : Calculator("default") {}

    Calculator(const std::string& name) : name(name) {
        history.reserve(10);
    }

    int add(int a, int b) // … method body omitted (6 lines)

    int multiply(int a, int b) // … method body omitted (5 lines)

    std::vector<std::string> getHistory() const {
        return history;
    }

private:
    bool validateInput(int value) // … method body omitted (17 lines)
};

ProcessingResult processUserData(const std::vector<User>& users) // … function body omitted (15 lines)

// Template function
template<typename T, typename Processor>
std::vector<T> processArray(const std::vector<T>& items, Processor processor) // … function body omitted (13 lines)

// Lambda usage function
std::vector<int> filterPositive(const std::vector<int>& numbers) // … function body omitted (6 lines)

int main() // … function body omitted (15 lines)
